src/Descriptor.hpp

```diff
@@ -107,7 +106,9 @@ class Descriptor {
     void set_endpoint(uint32_t netaddr, uint16_t port, std::string_view raw_full_hostname);
 
     [[nodiscard]] const std::string &host() const noexcept { return masked_host_; }
+    void host(std::string_view host) { masked_host_ = host; }
```

Comment:
Added to Descriptor as they're runtime state not tracked in Char itself, and I needed pfu to be able to retain the char's previous login from/time.